### ISSUE TITLE
usagePlan does not apply quota or throttle

### ISSUE THREAD

**Issue by dazza-codes** [serverless/serverless#9306]
The `serverless.yml` snippet below defines an API-Gateway `usagePlan`:

```yaml
provider:
  usagePlan:
    - demo:
      quota:
        limit: 500
        offset: 1
        period: DAY
      throttle:
        burstLimit: 20
        rateLimit: 5
```

But when the plan is viewed in the AWS Console for the API-Gateway created, the quota and throttle are unlimited.

The `.serverless/cloudformation-template-update-stack.json` contains something like:

```json
    "ApiGatewayUsagePlanDemo": {
      "Type": "AWS::ApiGateway::UsagePlan",
      "DependsOn": "ApiGatewayDeploymentxxx",
      "Properties": {
        "ApiStages": [
          {
            "ApiId": {
              "Ref": "ApiGatewayRestApi"
            },
            "Stage": "demo-prod"
          }
        ],
        "Description": "Usage plan \"demo\" for demo-api demo-prod stage",
        "UsagePlanName": "demo-api-demo-demo-prod"
      }
    },
```

<b>Installed version</b>

```
$ sls --version
Framework Core: 1.83.2
Plugin: 3.8.4
SDK: 2.3.2
Components: 2.34.9
```

**Comment by pgrzesik** [comment]
Hello @dazza-codes, thanks for reporting. First thing I've noticed is the invalid `yaml` configuration, it should be:

```
provider:
  usagePlan:
    - demo:
        quota:
          limit: 500
          offset: 1
          period: DAY
        throttle:
          burstLimit: 20
          rateLimit: 5
```

Additionally, you're not using the latest version of the Framework. Is there something preventing you from update?

**Comment by dazza-codes** [comment]
Thanks for the syntax correct, I will try to revise that.  I also noted in the docs a sample where the api-keys and the usage plan are nested below an `apiGateway` entity, i.e.

```yaml
provider:
  name: aws
  apiGateway:
    apiKeys:
      - myFirstKey
      - ${opt:stage}-myFirstKey
      - ${env:MY_API_KEY} # you can hide it in a serverless variable
      - name: myThirdKey
        value: myThirdKeyValue
      - value: myFourthKeyValue # let cloudformation name the key (recommended when setting api key value)
        description: Api key description # Optional
        customerId: A string that will be set as the customerID for the key # Optional
    usagePlan:
      quota:
        limit: 5000
        offset: 2
        period: MONTH
      throttle:
        burstLimit: 200
        rateLimit: 100
```

It seems to be working, to some extent, without nesting below an `apiGateway` entity.  But it might be worth also adding that.

I tried to update serverless a couple months ago and ran into weird validation errors that were reported as bugs on serverless at the time (and confirmed).  I've been flat out on product delivery deadlines since and not tempted to update until some kinks are worked out by the community.  It could be time to try to upgrade again.

**Comment by dazza-codes** [comment]
- corrected the indentation for the `quota` and `throttle` ✅ 
- tried to nest it all under `apiGateway` ❌ 

It now creates the CloudFormation details correctly:

```json

    "ApiGatewayUsagePlanDemo": {
      "Type": "AWS::ApiGateway::UsagePlan",
      "DependsOn": "ApiGatewayDeploymentxxx",
      "Properties": {
        "ApiStages": [
          {
            "ApiId": {
              "Ref": "ApiGatewayRestApi"
            },
            "Stage": "demo-prod"
          }
        ],
        "Description": "Usage plan \"demo\" for demo-api demo-prod stage",
        "UsagePlanName": "demo-api-demo-demo-prod",
        "Quota": {
          "Limit": 5000,
          "Offset": 1,
          "Period": "DAY"
        },
        "Throttle": {
          "BurstLimit": 20,
          "RateLimit": 5
        }
      }
    },
```

**Comment by pgrzesik** [comment]
Hello @dazza-codes :wave: Happy to hear you were able to resolve your issue. In order to nest the settings under `apiGateway` you'll need to upgrade the Framework - it was released with `2.18.0` - https://github.com/serverless/serverless/blob/master/CHANGELOG.md#2180-2021-01-07
